### PR TITLE
feat(components): visible pointer drag preview for Sortable/KanbanBoard (2252e84d)

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

Pointer-dragging a Sortable/KanbanBoard card gave no visible feedback (the shared `_sortable-item` uses pointer events, not native DnD, so there's no browser drag image).

Added a drag **preview portal** in the shared `_sortable-item.svelte`: on pointerdown it clones the row into a `position: fixed` `div[data-cinder-drag-preview]` on `document.body` that follows the pointer (updated each rAF frame via CSS custom properties, preserving the original grab offset); the source row becomes a dashed translucent **placeholder** marking the drop position. Both SortableList and KanbanBoard get the feedback without duplication. Drop, cancel, pointercancel, Escape (handle- and window-level), and component destroy all remove the portal and cancel the rAF — no orphaned overlay or frame.

Stayed with pointer events (no native-DnD switch). Keyboard drag + live-region announcements are untouched. Auto-scroll limitations (window-vertical only; horizontal/nested-scroll not implemented) are documented with guarding tests. Public API unchanged.

## Review committee

Codex (gpt-5.4, xhigh) + author. All 5 Codex findings on the shared-primitive portal were real and fixed: preview CSS now in both component stylesheets (a KanbanBoard subpath consumer was getting an unstyled body portal); cloned `id` attributes stripped + `inert` set on the portal (no duplicate-id a11y breakage); `destroyPreviewPortal` now cancels the rAF in every cleanup path (no orphaned frame leaking into a later drag); grab-offset preserved; and lifecycle regression tests added (Escape+queued-rAF, exact portal count idle/dragging, rapid second drag, clone-id-stripping).

## Test plan
- [x] `bun run --filter=cinder test -- kanban-board sortable-list component-css` — 133 pass
- [x] `components:check` / `typecheck` / `lint` (oxlint+stylelint) — clean
- Note: state-contract tested; pixel-level preview-follows-pointer needs a real browser.

---

> [!NOTE]
> **Medium Risk**
> Touches shared drag DOM (body portals, id stripping, rAF lifecycle) and Kanban pointer hit-testing; mitigated by contract tests and no public API changes, but regressions could affect reorder UX or a11y if cleanup paths miss an edge case.
> 
> **Overview**
> Adds **visible pointer-drag feedback** to the shared `_sortable-item` primitive used by **SortableList** and **KanbanBoard**: on pointer lift it appends a fixed `document.body` portal (`[data-cinder-drag-preview]`) that clones the row and tracks the pointer (grab offset + rAF updates); the in-list row becomes a dashed **placeholder** instead of the keyboard **lifted** styling.
> 
> **KanbanBoard** fixes pointer drop-target math by excluding the dragged card via stable `data-key` (not `--lifted`, which pointer drags no longer use). Preview/placeholder **CSS is duplicated** into `kanban-board.css` so Kanban-only style imports still position the body portal; a11y notes call out vertical-only auto-scroll limits.
> 
> Lifecycle hardening: portal teardown on drop/cancel/Escape/destroy, **move rAF cancelled** with the portal, rejected concurrent lifts, clone **`id` stripping** + `inert`/`aria-hidden` on the preview. Large **state-contract** test suites added; public API unchanged.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 06ea0f35f143444518fa20f7ea14d2a9101857ff. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 1]
```diff
@@ -22,6 +22,26 @@
...
+   * visualColumns / visualItems already re-inserts the placeholder at the
+   * destination index as the pointer moves.
+   */
+  .cinder-sortable-item--placeholder {
+    opacity: 0.4;
+    border: 2px dashed var(--cinder-border-muted, rgb(0 0 0 / 0.15));
+    border-radius: var(--cinder-radius-md, 4px);
+    box-shadow: none;
+    background: transparent;
+  }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css:44]: Fixed in 7c272cca: the placeholder no longer changes the row box size. SortableList uses outline:2px dashed (outside the box model) instead of adding a border; KanbanBoard keeps its 1px border width (border-color:transparent) + an outline for the dashed indicator. Box geometry is identical at rest and during drag, so no jitter and the drop-index getBoundingClientRect math is unaffected.

File: packages/components/src/components/kanban-board/kanban-board.css [hunk 2]
```diff
@@ -114,6 +114,18 @@
...
+   * During a pointer drag the card in the list becomes a placeholder showing
+   * the drop target position. The preview overlay (in sortable-list.css) is
+   * what follows the pointer; this keeps the drop slot visible in the column.
+   */
+  .cinder-kanban-board__card.cinder-sortable-item--placeholder {
+    min-block-size: var(--cinder-space-12, 3rem);
+    border: 2px dashed var(--cinder-ring-color, currentColor);
+    background: transparent;
+    box-shadow: none;
+  }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.css:135]: Fixed in 7c272cca: the placeholder no longer changes the row box size. SortableList uses outline:2px dashed (outside the box model) instead of adding a border; KanbanBoard keeps its 1px border width (border-color:transparent) + an outline for the dashed indicator. Box geometry is identical at rest and during drag, so no jitter and the drop-index getBoundingClientRect math is unaffected.

File: packages/components/src/components/sortable-list/sortable-list.test.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // @ts-nocheck — test file; noUncheckedIndexedAccess and bun:test types disabled per project convention
 /// <reference lib="dom" />
-import { describe, expect, mock, test } from 'bun:test';
+import { afterEach, beforeEach, describe, expect, mock, test } from 'bun:test';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.test.ts]: Fixed in 7c272cca: removed the unused beforeEach import and the dead baseline-body-children tracking block (it had no assertions).

File: packages/components/src/components/sortable-list/sortable-list.test.ts [hunk 4]
```diff
@@ -539,3 +539,421 @@ describe('SortableList', () => {
...
+}
+
+describe('SortableList pointer drag preview', () => {
+  // Capture the body's child count before each test so we can assert portal
+  // creation and removal relative to a known baseline.
+  let baselineBodyChildren: number;
+
+  beforeEach(() => {
+    baselineBodyChildren = document.body.children.length;
+  });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.test.ts]: Fixed in 7c272cca: removed the unused beforeEach import and the dead baseline-body-children tracking block (it had no assertions).

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 5]
```diff
@@ -60,4 +84,55 @@
...
+   * pointer drag. It clones the lifted item's content and follows the pointer
+   * so the user always has a clear visual of what they are dragging.
+   *
+   * Positioning uses CSS custom properties set by _sortable-item.svelte:
+   *   --preview-left / --preview-top   — row's bounding rect at lift time
+   *   --preview-dx / --preview-dy      — pointer delta since lift
+   *   --preview-width / --preview-height — row dimensions at lift time
+   *
+   * The transform translates the preview to the pointer location.
+   */
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed in 7c272cca: corrected the comment — the preview is positioned via inset-* custom properties; transform is used only for rotate/scale.

File: packages/components/src/components/_sortable-item.svelte [hunk 6]
```diff
@@ -57,6 +58,17 @@
   let latestPointerX = 0;
   let listEl: HTMLElement | null = null;
 
+  // Preview overlay state — reactive so the template can render the portal div.
+  // These only change during a pointer drag, never during keyboard lifts.
+  let previewX = $state(0);
+  let previewY = $state(0);
+  let previewWidth = $state(0);
+  let previewHeight = $state(0);
+  let isDraggingWithPointer = $state(false);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/_sortable-item.svelte]: Addressed in 7c272cca: previewWidth/previewHeight were demoted from $state to plain let (used only synchronously inside createPreviewPortal, never in reactive rendering). The portal is imperative by design — a document.body element that follows the pointer lives outside Svelte's reactive tree and is updated directly each rAF; added a comment noting which vars stay $state and why.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 7]
```diff
@@ -60,4 +84,55 @@
...
+  .cinder-sortable-drag-preview {
+    position: fixed;
+    inset-block-start: calc(var(--preview-top, 0px) + var(--preview-dy, 0px));
+    inset-inline-start: calc(var(--preview-left, 0px) + var(--preview-dx, 0px));
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed: switched the fixed-position preview portal from inset-inline-start/inset-block-start to physical left/top in both sortable-list.css and kanban-board.css. The coords come from getBoundingClientRect (physical viewport pixels), so physical left/top is correct and direction-independent — inset-inline-start would mirror to the right edge in RTL. (The in-flow placeholder keeps logical properties.)

File: packages/components/src/components/_sortable-item.svelte [hunk 8]
```diff
@@ -355,13 +482,17 @@
...
   data-sortable-row
   data-key={itemKey}
   data-row-id={rowId}
+  data-preview-x={isDraggingWithPointer ? previewX : undefined}
+  data-preview-y={isDraggingWithPointer ? previewY : undefined}
   aria-roledescription="sortable item"
   class={cn(
     'cinder-sortable-item',
-    isLifted && 'cinder-sortable-item--lifted',
+    isLifted && isDraggingWithPointer && 'cinder-sortable-item--placeholder',
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/_sortable-item.svelte:510]: Real + HIGH — fixed in b58a9ad3. locatePointerTarget excluded the dragged row by the class --lifted, but pointer drags now use --placeholder (only keyboard uses --lifted), so the dragged card wasn't excluded from the midpoint calc → off-by-one insertion. Switched the filter to the stable data-key attribute (present in both drag states), decoupling drop logic from the visual class. Added tests for the class-state invariant in both modes.

File: packages/components/src/components/kanban-board/kanban-board.css [hunk 9]
```diff
@@ -114,6 +114,25 @@
...
+   */
+  .cinder-kanban-board__card.cinder-sortable-item--placeholder {
+    min-block-size: var(--cinder-space-12, 3rem);
+    border-color: transparent;
+    outline: 2px dashed var(--cinder-ring-color, currentColor);
+    outline-offset: -1px;
+    background: transparent;
+    box-shadow: none;
+  }
+
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.css:136]: Fixed in b58a9ad3: added opacity:0.4 + > * { visibility: hidden } to the kanban placeholder rule so a cinder/kanban-board/styles-only consumer no longer sees full card content duplicated in the placeholder + preview — matches the sortable-list placeholder behavior.

File: packages/components/src/components/kanban-board/kanban-board.css [hunk 10]
```diff
@@ -114,6 +114,25 @@
...
+   * skewing the midpoint geometry used by drop-target calculations.
+   */
+  .cinder-kanban-board__card.cinder-sortable-item--placeholder {
+    min-block-size: var(--cinder-space-12, 3rem);
+    border-color: transparent;
+    outline: 2px dashed var(--cinder-ring-color, currentColor);
+    outline-offset: -1px;
+    background: transparent;
+    box-shadow: none;
+  }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.css:135]: Fixed in b58a9ad3: added opacity:0.4 + > * { visibility: hidden } to the kanban placeholder rule so a cinder/kanban-board/styles-only consumer no longer sees full card content duplicated in the placeholder + preview — matches the sortable-list placeholder behavior.

File: packages/components/src/components/sortable-list/sortable-list.a11y.md [hunk 11]
```diff
@@ -49,6 +49,18 @@ HTML5 Drag and Drop API is **not used**. iOS Safari does not fire touch events o
...
 
+## Drag Preview
+
+During a pointer drag a fixed-position overlay (`.cinder-sortable-drag-preview`) is appended to `document.body`. The overlay clones the lifted row's HTML so the user can see what they are dragging as the pointer moves. The preview:
+
+- Has `aria-hidden="true"` — it is purely visual and does not duplicate AT content.
+- Has `pointer-events: none` — it does not interfere with drop-target hit testing.
+- Is removed on drop, cancel, `pointercancel`, Escape (both handle-level and window-level), and component destroy.
+
+The source row switches from `.cinder-sortable-item--lifted` to `.cinder-sortable-item--placeholder` during a pointer drag. The placeholder uses a dashed border and reduced opacity to mark the current drop target position without showing full card content.
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.a11y.md]: Fixed in b58a9ad3: corrected 'dashed border' -> 'dashed outline' (outline avoids box-size change) and documented that the placeholder hides its content (visibility:hidden), showing drop position only.

File: packages/components/src/components/kanban-board/kanban-board.a11y.md [hunk 12]
```diff
@@ -10,6 +10,14 @@ Collapsed columns keep their headers, column handles, and collapse controls reac
 
 The component does not use deprecated `aria-grabbed` or `aria-dropeffect`. Announcements are sent through one live region and include the moved card or column, destination, position, drop, cancellation, and collapse or expand state.
 
-Pointer dragging uses the shared cinder sortable handle implementation and recomputes board column geometry during the gesture. Nested scroll-container auto-scroll is out of scope for this version; window-edge auto-scroll is inherited from the shared sortable handle behavior.
+Pointer dragging uses the shared cinder sortable handle implementation and recomputes board column geometry during the gesture.
+
+During a pointer drag a fixed-position overlay (`.cinder-sortable-drag-preview`, appended to `document.body`) shows what is being dragged. The card's source row becomes a placeholder (dashed border, reduced opacity) marking the current drop insertion position. Both the overlay and the placeholder class are removed on drop, cancel, or any cleanup path.
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.a11y.md]: Fixed in b58a9ad3: corrected 'dashed border' -> 'dashed outline' (outline avoids box-size change) and documented that the placeholder hides its content (visibility:hidden), showing drop position only.

File: packages/components/src/components/sortable-list/sortable-list.css [hunk 13]
```diff
@@ -60,4 +90,59 @@
...
+   * always returns physical viewport distances regardless of writing direction.
+   * Using inset-inline-start would map to the right edge in RTL layouts, causing
+   * the preview to appear on the wrong side of the viewport.
+   */
+  .cinder-sortable-drag-preview {
+    position: fixed;
+    top: calc(var(--preview-top, 0px) + var(--preview-dy, 0px));
+    left: calc(var(--preview-left, 0px) + var(--preview-dx, 0px));
+    inline-size: var(--preview-width, auto);
+    block-size: var(--preview-height, auto);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/sortable-list/sortable-list.css]: Fixed: switched the preview portal sizing from inline-size/block-size to physical width/height (the values come from getBoundingClientRect().width/.height, physical pixels), consistent with the physical left/top positioning. Both stylesheets updated.

File: packages/components/src/components/_sortable-item.svelte [hunk 14]
```diff
@@ -184,6 +303,11 @@
 
     handleEl.setPointerCapture(event.pointerId);
     context.lift(itemKey, index, itemLabel, total);
+
+    // Create the drag preview after lift so the row is in lifted state
+    // when we clone it, then position relative to the pointer.
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/_sortable-item.svelte:324]: Fixed: handlePointerDown now checks controller.phase/liftedKey after context.lift() and bails (releasing pointer capture) if the lift was a no-op (another item already lifted), so no orphan portal is created. Added a regression test (concurrent pointerdown keeps portal count at 1).

File: packages/components/src/components/kanban-board/kanban-board.test.ts [hunk 15]
```diff
@@ -603,3 +603,352 @@ describe('KanbanBoard', () => {
...
+    await fireEvent.pointerUp(handle, { pointerId: 1, pointerType: 'mouse' });
+
+    // After drop, the card row is at its new logical position. Query by key.
+    const cardRow = container.querySelector('[data-key="a"]') as HTMLElement;
+    if (cardRow) {
+      expect(cardRow.classList.contains('cinder-sortable-item--placeholder')).toBe(false);
+      expect(cardRow.hasAttribute('data-preview-x')).toBe(false);
+      expect(cardRow.hasAttribute('data-preview-y')).toBe(false);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.test.ts:779]: Fixed: replaced the if(cardRow) guard with expect(cardRow).not.toBeNull() before the assertions, so a missing row after a no-move drop fails the test instead of passing vacuously.

File: packages/components/src/components/kanban-board/kanban-board.css [hunk 16]
```diff
@@ -114,6 +114,31 @@
...
+   */
+  .cinder-kanban-board__card.cinder-sortable-item--placeholder {
+    min-block-size: var(--cinder-space-12, 3rem);
+    opacity: 0.4;
+    border-color: transparent;
+    outline: 2px dashed var(--cinder-ring-color, currentColor);
+    outline-offset: -1px;
+    background: transparent;
+    box-shadow: none;
+  }
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/kanban-board/kanban-board.css:135]: Fixed: added the matching @media (prefers-reduced-motion: reduce) opacity override (0.6) to the kanban placeholder rule, for parity with sortable-list.css.
